### PR TITLE
[DOCS] Add delete snapshot repo API docs

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:

* Adds API reference docs for the delete snapshot repo API.

* Corrects an error in the delete snapshot repo API spec. Comma-separated repository names are not supported.

* Relocates the existing delete snapshot repo API example docs.

### CODE REVIEW

File: docs/reference/snapshot-restore/apis/delete-repo-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
+[[delete-snapshot-repo-api]]
+=== Delete snapshot repository API
+++++
+<titleabbrev>Delete snapshot repository</titleabbrev>
+++++
+
+Deletes one or more <<snapshots-register-repository,snapshot repositories>>.
```
Comment (DaveCTurner) [inline docs/reference/snapshot-restore/apis/delete-repo-api.asciidoc:7]: TIL we accept wildcards here :) But, trappily, not a comma-separated list.

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
The other docs of this API are on the "take a snapshot" page for, I think, historical reasons. Could you move them to the "register a repository" page? Looks like just above the "shared file system repository" heading would be a good spot.

**Comment by jrodewig** [comment]
Thanks @DaveCTurner. I've relocated the other delete repo docs to "Register a snapshot." I also updated some of the wording across the docs to better clarify that the delete snapshot API doesn't delete snapshots.

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Review by DaveCTurner — APPROVED** [review]
LGTM thanks @jrodewig

**Comment by jrodewig** [comment]
### Backport commits
master eae4a1c
7.x b3426dd 
7.8 0e73721 
7.7 e68a2ec

**Pull request merged**